src/openrct2/ride/Vehicle.cpp

```diff
@@ -3640,9 +3640,8 @@ void Vehicle::UpdateCollisionSetup()
 
         train->IsCrashedVehicle = true;
         train->animationState = scenario_rand();
-        train->var_CA = scenario_rand();
 
-        train->animation_frame = train->var_CA & 0x7;
+        train->animation_frame = (train->animationState >> 16) & 0x7;
```

Comment:
This is a bit odd. It should be refactored. Will need to understand the rest of the users of this random information but id prefer if this was just `scenario_rand() & 0x7`

Reply:
> I believe the only user of that information is the other function you've highlighted. It chooses a random frame of 8 frames of coaster debris while spawning the coaster debris, then in the other function it animates them.

Yes my conclusion is the same. I think the best solution to this then is the following for this bit of code:
```
train->animationState = sceanrio_rand() & 0xFFFF;
train->animation_frame = sceanrio_rand() & 0x7;
```
That means that the animation frame will not immediately change on the next tick due to being >58248 in the vast majority of cases.